static/js/narrow.js

```diff
@@ -795,7 +795,29 @@ function pick_empty_narrow_banner() {
     var num_operators = current_filter.operators().length;
 
     if (num_operators !== 1) {
-        // For multi-operator narrows, we just use the default banner
+        // For invalid-multi-operator narrows, we display an invalid narrow banner
+
+        var senders = _.filter(current_filter.operators(), function (term) {
+            return term.operator === "sender";
+        });
+
+        // No message can have multiple streams
+        if (_.filter(current_filter.operators(), function (term) {
+            return term.operator === "stream";
+        }).length > 1 ||
+        // No message can have multiple topics
+        _.filter(current_filter.operators(), function (term) {
+            return term.operator === "topic";
+        }).length > 1 ||
+        // No message can have multiple senders
+        senders.length > 1) {
+            return $("#invalid_narrow_message");
```

Comment:
Could this be done much less messily with e.g. `current_filter.operands("sender").length > 1)`?